### PR TITLE
Refactor mapping

### PR DESCRIPTION
**Pull request by sgratzl**
generalize the mapping manager approach to also handle simple mapping file cases

### CODE REVIEW

File: phovea_server/dataset_csv.py [hunk 1]
```diff
@@ -131,11 +131,10 @@ def cmp(a, b):
     clusters = [dict(name=k, range=clusters.get(k, []), color=colors.get(k, 'gray')) for k in groups]
 
     rows = np.array([di[0] for di in data[1:]])
-    return {
-        'rows': rows,
-        'rowIds': assign_ids(rows, self.idtype),
-        'groups': clusters
-    }
+    return {'rows': rows,
```
Comment (thinkh) [inline phovea_server/dataset_csv.py]: Why are return values (objects) now formated like that?

**Comment by sgratzl** [comment]
> How can I see and/or test the new mapper?

the client api hasn't changed: https://github.com/phovea/phovea_core/blob/master/src/idtype.ts#L211

**Comment by thinkh** [comment]
@chrilehner Could you please check out this `refactor_mapping` branch and try if the mapping still works with Ordino? If yes, then I can merge this PR.

**Comment by chrilehner** [comment]
I checked out the branch and tried to:
- load a named gene set and cellline panel and open different views (e.g. co-expression, oncoprint, ...)
- import files and open different views
- search for genes in the single entity entrypoint search and open different views

the only problems I saw were that in some views the server was not found, but the mapping worked for me

anything else I need to check?

**Comment by thinkh** [comment]
@chrilehner Thanks for testing. No, I think that should be fine.

**Comment by chrilehner** [comment]
you're welcome, @thinkh

**Pull request merged**